internal/pkg/variable/parser.go

```diff
@@ -1,268 +1,20 @@
-// Copyright (c) HashiCorp, Inc.
-// SPDX-License-Identifier: MPL-2.0
-
 package variable
 
 import (
-	"errors"
-	"fmt"
-	"os"
-	"sort"
-	"strings"
-
 	"github.com/hashicorp/hcl/v2"
-	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/nomad-pack/internal/pkg/variable/parser"
+	"github.com/hashicorp/nomad-pack/internal/pkg/variable/parser/config"
 	"github.com/hashicorp/nomad-pack/sdk/pack"
-	"github.com/spf13/afero"
-	"github.com/zclconf/go-cty/cty"
 )
 
-// Parser can parse, merge, and validate HCL variables from multiple different
-// sources.
-type Parser struct {
-	fs  afero.Afero
-	cfg *ParserConfig
-
-	// rootVars contains all the root variable declared by all parent and child
-	// packs that are being parsed. The first map is keyed by the pack name,
-	// the second is by the variable name.
-	rootVars map[string]map[string]*Variable
-
-	// envOverrideVars, fileOverrideVars, cliOverrideVars are the override
-	// variables. The maps are keyed by the pack name they are associated to.
-	envOverrideVars  map[string][]*Variable
-	fileOverrideVars map[string][]*Variable
-	cliOverrideVars  map[string][]*Variable
+type Parser interface {
```

Comment:
I'm not sure about the usefulness of this interface. In the code we only ever use it once: in the `NewParser` constructor as its return type, but then I wonder: as a caller of a new parser constructor, wouldn't I want to be able to assert the type of the parser (v1/v2)? Perhaps there's something I'm missing here but I feel this interface is just superfluous.